test/cmdlineTests/asm_json_export_yul_with_debug/args

```diff
@@ -0,0 +1 @@
+--strict-assembly - --asm-json
```

Comment:
Let's pretty print the output in these tests. It's not great that it's that long (would be great if it could be reduced), but having it squashed into a single line is even worse and makes it very hard to actually review.

```suggestion
--strict-assembly - --asm-json --pretty-json --json-indent 4
```

Reply:
Also, the `asm_json_export_yul_with_debug/` does not follow the current naming convention (which will be annoying if we ever have to add similar tests for non-Yul input and for Standard JSON). I'd name this one `strict_asm_asm_json_with_debug_info_annotations/` and others in a similar vein.

Reply:
We will have really a lot of extra lines with these tests. maybe I can minimize them a bit.

Reply:
ok I rename all the `asm_json_export_yul_` to `strict_asm_asm_json_`

Reply:
See #14866.

Reply:
I added pretty printing and also cut out a bit of code so it's a lot shorter.

And now you can actually see something in there. When it's all bunched up it's hard to even verify the output is correct.